postgres/manage.sls

```diff
@@ -44,7 +44,25 @@ postgres-reload-modules:
 # Database states
 
 {%- for name, db in postgres.databases|dictsort() %}
-
+  {%- if 'extensions' in db %}
+    {%- for ext_name, extension in db.pop('extensions')|dictsort() %}
+      {%- do extension.update({'maintenance_db': name }) %}
+{{ format_state( name + '-' + ext_name, 'postgres_extension', extension) }}
+    - require:
+      - postgres_database: postgres_database-{{ name }}
+      {%- if 'schema' in extension %}
+      - postgres_schema: postgres_schema-{{ extension.schema }}
+      {%- endif %}
+    {%- endfor %}
+  {%- endif %}
+  {%- if 'schemas' in db %}
+    {%- for schema_name, schema in db.pop('schemas')|dictsort() %}
+      {%- do schema.update({'dbname': name }) %}
+{{ format_state( schema_name, 'postgres_schema', schema) }}
+    - require:
+      - postgres_database: postgres_database-{{ name }}
+    {%- endfor %}
+  {%- endif %}
```

Comment:
Following the logic of raw SQL command, we should first create a database, then schema and finally install an existing extension. You could pop up per-DB `schemas` and `extensions` parameters from Pillar dictionary for a single database ahead, and then construct the states afterwards in a logical order.

Reply:
Logic is OK. States are executed in correct order. Arrangement in file is irrelevant.

Reply:
Yep. Requisite parameters simply put them in right order. That's OK.